h/api/schemas.py

```diff
@@ -256,7 +256,7 @@ def validate(self, data):
         # group they've asked to create one in.
         if 'group' in appstruct:
             if appstruct['group'] == '__world__':
-                group_principal = security.Everyone
+                group_principal = security.Authenticated
```

Comment:
Users need to be in `security.Authenticated` to create an annotation in `__world__`, not just in `security.Everyone`, which includes unauthenticated users. This change is necessary so that [this test](https://github.com/hypothesis/h/pull/3333/files#diff-bb0455f689fc16e7f1d9fc5556c04bc5R436) continues to pass.